pkg/arvo/sys/vane/gall.hoon

```diff
@@ -493,10 +502,49 @@
   ::
   ++  mo-retrieve-duct
     |=  [=ship index=@ud]
-    ^-  duct
+    ^-  (unit duct)
     ::
-    =/  =foreign  (~(got by contacts.agents.state) ship)
-    (~(got by duct-map.foreign) index)
+    =/  contact=(unit foreign)  (~(get by contacts.agents.state) ship)
+    ?~  contact
+      ~
+    `(~(got by duct-map.u.contact) index)
```

Comment:
My instincts cry out for a bind (er, a `biff`) but I don't think that's actually clearer here.  Alas.